libtiledbsoma/src/soma/soma_dataframe.h

```diff
@@ -52,54 +53,73 @@ class SOMADataFrame : public SOMAObject {
     /**
      * @brief Create a SOMADataFrame object at the given URI.
      *
-     * @param ctx TileDB context
      * @param uri URI to create the SOMADataFrame
      * @param schema TileDB ArraySchema
-     * @return std::unique_ptr<SOMADataFrame> opened in read mode
+     * @param platform_config Optional config parameter dictionary
+     * @return std::shared_ptr<SOMADataFrame> opened in read mode
+     */
+    static std::shared_ptr<SOMADataFrame> create(
+        std::string_view uri,
+        ArraySchema schema,
+        std::map<std::string, std::string> platform_config = {});
+
+    /**
+     * @brief Create a SOMADataFrame object at the given URI.
+     *
+     * @param uri URI to create the SOMADataFrame
+     * @param schema TileDB ArraySchema
+     * @param ctx TileDB context
+     * @return std::shared_ptr<SOMADataFrame> opened in read mode
      */
-    static std::unique_ptr<SOMADataFrame> create(
-        std::shared_ptr<Context> ctx, std::string_view uri, ArraySchema schema);
+    static std::shared_ptr<SOMADataFrame> create(
+        std::string_view uri, ArraySchema schema, std::shared_ptr<Context> ctx);
 
     /**
      * @brief Open and return a SOMADataFrame object at the given URI.
      *
-     * @param mode TILEDB_READ or TILEDB_WRITE
+     * @param mode read or write
      * @param uri URI to create the SOMADataFrame
      * @param column_names A list of column names to use as user-defined index
      * columns (e.g., ``['cell_type', 'tissue_type']``). All named columns must
      * exist in the schema, and at least one index column name is required.
      * @param platform_config Platform-specific options used to create this
      * DataFrame
+     * @param result_order Read result order: automatic (default), rowmajor, or
+     * colmajor
      * @param timestamp If specified, overrides the default timestamp used to
      * open this object. If unset, uses the timestamp provided by the context.
-     * @return std::unique_ptr<SOMADataFrame> SOMADataFrame
+     * @return std::shared_ptr<SOMADataFrame> SOMADataFrame
      */
-    static std::unique_ptr<SOMADataFrame> open(
-        tiledb_query_type_t mode,
+    static std::shared_ptr<SOMADataFrame> open(
         std::string_view uri,
```

Comment:
As noted by @teo-tsirpanis in the other instance, this is an interface change so maybe I should check that it doesn't affect me.

Reply:
I have changed this back to `unique_ptr`.